ui/app/src/components/GlobalConfigManagement/GlobalConfigManagement.tsx

```diff
@@ -86,14 +76,52 @@ export function GlobalConfigManagement() {
     ];
 
     forkJoin(requests).subscribe(([sample, content]) => {
+      const sessionContent = sessionStorage.getItem(sessionStorageKey);
+
       setLastSavedContent(content);
-      setContent(content);
-      setContentSize(content.length);
+      if (sessionContent) {
+        setContent(sessionContent);
+        setContentSize(sessionContent.length);
+        setHasChanges(true);
+        sessionStorage.removeItem(sessionStorageKey);
+      } else {
+        setContent(content);
+        setContentSize(content.length);
+      }
       setSample(sample);
       setEnable(false);
     });
   });
 
+  useUnmount(() => {
+    if (hasChangesRef.current) {
+      sessionStorage.setItem(sessionStorageKey, aceEditorRef.current.getValue());
+      const eventId = 'unsavedConfigManagementChangesConfirmation';
+      dispatch(
+        showConfirmDialog({
+          body: <FormattedMessage defaultMessage="You left unsaved changes. Go back and continue editing?" />,
+          onCancel: batchActions([closeConfirmDialog(), dispatchDOMEvent({ id: eventId, button: 'cancel' })]),
+          onOk: batchActions([closeConfirmDialog(), dispatchDOMEvent({ id: eventId, button: 'ok' })]),
+          okButtonText: <FormattedMessage defaultMessage="Continue editing" />,
+          cancelButtonText: <FormattedMessage defaultMessage="Discard changes" />
+        })
+      );
+      createCustomDocumentEventListener<{ button: 'ok' | 'cancel' }>(eventId, ({ button }) => {
+        if (button === 'ok') {
+          navigate('/global-config');
```

Comment:
Use the enum.

Look at other places we might be using routes like Launcher and its subcomponents, etc